TSPL.docc/LanguageGuide/ErrorHandling.md

```diff
@@ -699,6 +699,200 @@ let photo = try! loadImage(atPath: "./Resources/John Appleseed.jpg")
   ```
 -->
 
+## Specifying the Error Type
+
+All of the examples above use the most common kind of error handling,
+where the errors that your code throws
+can be values of any type that conforms to the `Error` protocol.
+This approach matches the reality that
+you don't know ahead of time every error that could happen
+while the code is running,
+especially when propagating errors thrown somewhere else.
+This approach also reflects the fact that errors can change over time.
+New versions of a library can throw new errors ---
+including libraries used by your dependencies ---
+and the rich complexity of real-world user configurations
+can expose failure modes that weren't visible during development or testing.
+The error handling code in the examples above
+always includes a default case to handle errors
+that don't have their own specific `catch` clause.
+
+Most Swift code doesn't specify the type for the errors it throws.
+However,
+in some special cases,
+you might limit code to throwing errors of only one specific type:
+
+- When running code on an embedded system
+  that doesn't support dynamic allocation of memory.
+  Throwing an instance `any Error` or another boxed protocol type
+  requires allocating memory at runtime to store the error.
+  Throwing an error of a specific type
+  lets Swift allocate that memory upfront instead.
+
+- When the errors are used only within some unit of code,
```

Comment:
This is going to be a bit tricky to put into active voice, and not necessarily clearer — this is a good example of an actor that's poorly defined and unimportant.  It might be better as "When the errors are *contained* within some unit of code"?  Or to rewrite more aggressively:

> When the errors are an implementation detail of some unit of code,
> like a library,
> and aren't part of the interface to that code.

I'll start with that last rewrite, and we can iterate.